src/components/tests/tour.test.tsx

```diff
@@ -0,0 +1,228 @@
+import { createMount, createShallow } from '@material-ui/core/test-utils';
+import React from 'react';
+import DashboardTour from '../tour/tour';
+import { SpotlightValue } from '../spotlight-value';
+import { ReactWrapper } from 'enzyme';
+import { tourText } from '../tour/tour-data';
+
+/*react-leaflet does not work well on jsdom,
+suppress warnings of not being able to find elements from leaflet*/
+jest.spyOn(console, 'warn').mockImplementation(() => {});
+
+const mount = createMount();
+const shallow = createShallow();
+
+describe('Dashboard Tour', () => {
+  const tourState: boolean = true;
+  const setTourState: jest.Mock = jest.fn();
+  const setShowSettings: jest.Mock = jest.fn();
+  const setShowOmniPanel: jest.Mock = jest.fn();
+  const setShowHelp: jest.Mock = jest.fn();
+  const clearSpotlights: jest.Mock = jest.fn();
+  const setCurrentView: jest.Mock = jest.fn();
+  const doorSpotlight: SpotlightValue<string> = { value: 'main_door' };
+  const setDoorSpotlight: jest.Mock = jest.fn();
+
+  let root: ReactWrapper;
+  let expectedText: string;
+  let stepText: string;
+  let nextBtn: ReactWrapper;
+  const tourProps = {
+    tourState,
+    setTourState,
+    setShowSettings,
+    setShowOmniPanel,
+    setShowHelp,
+    clearSpotlights,
+    setCurrentView,
+    doorSpotlight,
+    setDoorSpotlight,
+  };
+  const wait = (delay = 0) => new Promise((resolve) => setTimeout(resolve, delay));
+
+  beforeEach(() => {
+    root = mount(<DashboardTour tourProps={tourProps} />);
+    nextBtn = root.find('[data-testid="next-btn"]').first();
+  });
+
+  afterEach(() => {
+    root?.unmount();
+  });
+
+  it('renders without crashing', () => {
+    const tour = shallow(<DashboardTour tourProps={tourProps} />);
+    expect(tour).toMatchSnapshot();
+  });
+
+  it('renders first welcome step', () => {
+    expectedText = tourText.step1Welcome;
+    stepText = root.find('[data-testid="stepBox"]').first().text();
+
+    expect(stepText.includes(expectedText));
+  });
+
+  it('renders zoom button step', async () => {
+    expectedText = tourText.step2Zoom;
+    nextBtn.simulate('click');
+    await wait(5);
+
+    stepText = root.find('[data-testid="stepBox"]').first().text();
+
+    expect(stepText.includes(expectedText));
+  });
+
+  it('renders floor plan button step', async () => {
+    expectedText = tourText.step3Floorplan;
+
+    for (let i = 1; i < 3; i++) {
```

Comment:
Sorry Val, this occurred to me now and I think it will be extremely useful for anyone who has to add more cards to the tutorial. Could it be that we could find a way not to deal with fixed numbers within the tests, because if, for example, I want to add a card in position 3 I will have to re-enumerate the tests> 4. E.g .:
10 11 12 13 14
10 (new) 11 12 13 14 15.

So can we somehow get the positions from `tourSteps` or` tourContent`. And we do something like this

for (let i = 1; i < tourStepts.Floorplan; i++)

where tourStepts.Floorplan === 3.

The same thing would go with the variable names I think


What do you think about it?

Reply:
hm it's a bit tricky because how the tour does it is that the step number can only be accessed through the 'content' variable... let me think about it, I know where you're coming from and I think that would be mighty helpful in the future as well, so thanks for bringing that up too!!